framework/src/problems/FEProblemBase.C

```diff
@@ -5329,6 +5329,164 @@ FEProblemBase::computeResidual(const NumericVector<Number> & soln, NumericVector
   computeResidualInternal(soln, residual, _fe_vector_tags);
 }
 
+void
+FEProblemBase::computeResidualAndJacobian(const NumericVector<Number> & soln,
```

Comment:
same thing here; I fear a maintenance burden. but at the same time... may be impossible to clean up something like this as the residual/jacobian method madness in FEProblemBase is insane 😆

Reply:
Yea this one is even tougher in my mind ... I honestly feel like there may be some inefficiencies in the current methods. Or more importantly, there are enough differences between the `computeResidual` and `computeJacobian` paths that without starting from scratch, working with objects for whom I know how they work, I fear I would miss some nuanced difference in how the residual and Jacobian have to be treated that could cause some difficult bugs to track down. Thus far I have mostly stuck with AD objects for whom I understand clearly how they work